File: day1/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	f, err := os.Open("./input.txt")
	if err != nil {
		fmt.Println(err)
	}

	numIncreased, prevNum := 0, 0
	numbers := []int{}
	scanner := bufio.NewScanner(f)
	for scanner.Scan() {
		num, err := strconv.Atoi(scanner.Text())
		if err != nil {
			fmt.Println(err)
		}

		if num > prevNum && prevNum != 0 {
			numIncreased++
		}

		prevNum = num
		numbers = append(numbers, num)
	}

	fmt.Printf("Number increased: %d\n", numIncreased)

	numIncreased = 0
	prevSum := 0
	for i := 0; i < len(numbers)-2; i++ {
		num := numbers[i] + numbers[i+1] + numbers[i+2]

		if num > prevSum && prevSum != 0 {
			numIncreased += 1
		}

		prevSum = num
	}
	fmt.Printf("Num of larger sums: %d\n", numIncreased)
}

File: day1/src/main.rs
use std::fs;

fn main() {
    let f = fs::read_to_string("./input.txt").expect("Unable to open file!");
    let lines = f.split('\n').collect::<Vec<&str>>();

    let mut prev_num = 0;
    let mut num_increased = 0;
    for &i in &lines {
        let num = i.parse::<i32>().unwrap();

        if num.gt(&prev_num) && !prev_num.eq(&0) {
            num_increased += 1;
        }
        prev_num = num;
    }

    println!("Part 1: Number increased: {}", num_increased);

    let mut prev_sum = 0;
    num_increased = 0;
    for i in 0..lines.len() - 2 {
        let num = lines[i].parse::<i32>().unwrap()
            + lines[i + 1].parse::<i32>().unwrap()
            + lines[i + 2].parse::<i32>().unwrap();
        if num.gt(&prev_sum) && !prev_sum.eq(&0) {
            num_increased += 1;
        }
        prev_sum = num;
    }

    println!("Part 2: Number increased: {}", num_increased);
}
